File: FestivalJava/Networking/src/main/java/rpcprotocol/Response.java
package rpcprotocol;

import java.io.Serializable;

/**
 * Created by Sebi on 29-Mar-17.
 */
public class Response implements Serializable {
    //type
    private ResponseType type;

    //data
    private Object data;

    private Response(){}

    //getters and setters
    public ResponseType getType() {
        return type;
    }

    public void setType(ResponseType type) {
        this.type = type;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    @Override
    public String toString(){
        return "Response { type = " + getType() + " data = " + getData() + " } ";
    }

    public static class Builder{
        private Response response = new Response();

        public Builder type(ResponseType type){
            response.setType(type);
            return this;
        }

        public Builder data(Object data){
            response.setData(data);
            return this;
        }

        public Response build(){
            return response;
        }
    }
}

File: FestivalC#/Festival/Networking/rpcprotocol/Response.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Networking.rpcprotocol
{
    [Serializable]
    public class Response
    {
        //type of response
        private ResponseType type;

        //data sent
        private object data;

        public ResponseType Type
        {
            get
            {
                return type;
            }

            set
            {
                type = value;
            }
        }

        public object Data
        {
            get
            {
                return data;
            }

            set
            {
                data = value;
            }
        }

        public Response(ResponseType type, object data)
        {
            this.type = type;
            this.data = data;
        }

        public override string ToString()
        {
            return "Response{ type = " + Type + " data = " + Data + " } ";
        }

    }
}
